access/handler.go

```diff
@@ -1179,16 +1252,29 @@ func (h *Handler) blockHeaderResponse(header *flow.Header, status flow.BlockStat
 }
 
 // buildMetadataResponse builds and returns the metadata response object.
-func (h *Handler) buildMetadataResponse() *entities.Metadata {
+// Expected error returns during normal operations:
+// - codes.Internal if an error is encountered getting the highest indexed height from indexer.
+func (h *Handler) buildMetadataResponse() (*entities.Metadata, error) {
 	lastFinalizedHeader := h.finalizedHeaderCache.Get()
 	blockId := lastFinalizedHeader.ID()
 	nodeId := h.me.NodeID()
 
-	return &entities.Metadata{
+	metadata := &entities.Metadata{
 		LatestFinalizedBlockId: blockId[:],
 		LatestFinalizedHeight:  lastFinalizedHeader.Height,
 		NodeId:                 nodeId[:],
 	}
+
+	if h.indexReporter != nil {
+		var err error
+		highestIndexedHeight, err := h.indexReporter.HighestIndexedHeight()
+		if err != nil {
+			return nil, rpc.ConvertIndexError(err, lastFinalizedHeader.Height, "could not get highest indexed height")
```

Comment:
this will result in all API requests failing until the index is initialized, which can take over an hour on the initial load.

Instead of returning an error in this case, I think this should check for the sentinel, and return `0` if the index is not initialized, or an error for any other failure.

I think `0` makes the most sense for a default value since:
* it's obvious to a human that it's not a recent block height
* a naive integer comparison is likely to return the correct answer if used to check if a block has been indexed (0 < any block therefore it's not indexed yet)